libs/server/Resp/KeyAdminCommands.cs

```diff
@@ -2,14 +2,228 @@
+
+            RawStringInput input;
+            if (expiry > 0)
+            {
+                var inputArg = DateTimeOffset.UtcNow.Ticks + TimeSpan.FromSeconds(expiry).Ticks;
+                input = new RawStringInput(RespCommand.SETEXNX, ref parseState, arg1: inputArg);
+            }
+            else
+            {
+                input = new RawStringInput(RespCommand.SETEXNX, ref parseState);
+            }
+
+            var status = storageApi.SET_Conditional(ref sbKey, ref input);
+
+            if (status is GarnetStatus.NOTFOUND)
+            {
+                while (!RespWriteUtils.WriteDirect(CmdStrings.RESP_OK, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_BUSSYKEY, ref dcurr, dend))
+                SendAndReset();
+
+            return true;
+        }
+
+        /// <summary>
+        /// DUMP 
+        /// </summary>
+        bool NetworkDUMP<TGarnetApi>(ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            if (parseState.Count != 1)
+            {
+                return AbortWithWrongNumberOfArguments(nameof(RespCommand.DUMP));
+            }
+
+            var key = parseState.GetArgSliceByRef(0);
+
+            var status = storageApi.GET(key, out var value);
+
+            if (status is GarnetStatus.NOTFOUND)
+            {
+                while (!RespWriteUtils.WriteDirect(CmdStrings.RESP_ERRNOTFOUND, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            Span<byte> encodedLength = stackalloc byte[5];
+
+            if (!RespLengthEncodingUtils.TryWriteLength(value.ReadOnlySpan.Length, encodedLength, out var bytesWritten))
+            {
+                while (!RespWriteUtils.WriteError("ERR DUMP payload length is invalid", ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            encodedLength = encodedLength.Slice(0, bytesWritten);
+
+            // Len of the dump (payload type + redis encoded payload len + payload len + rdb version + crc64)
+            var len = 1 + encodedLength.Length + value.ReadOnlySpan.Length + 2 + 8;
+            Span<byte> lengthInASCIIBytes = stackalloc byte[NumUtils.NumDigitsInLong(len)];
+            var lengthInASCIIBytesLen = NumUtils.LongToSpanByte(len, lengthInASCIIBytes);
+
+            // Total len (% + length of ascii bytes + CR LF + payload type + redis encoded payload len + payload len + rdb version + crc64 + CR LF)
+            var totalLength = 1 + lengthInASCIIBytesLen + 2 + 1 + encodedLength.Length + value.ReadOnlySpan.Length + 2 + 8 + 2;
+
+            byte[] rentedBuffer = null;
+            var buffer = totalLength <= 256
+                ? stackalloc byte[256]
+                : (rentedBuffer = ArrayPool<byte>.Shared.Rent(totalLength));
```

Comment:
You can directly write to the output buffer for non-large objects.

```suggestion
            var buffer = totalLength <= (dend - dcurr)
                ? new Span<byte>(dcurr, (int)(dend - dcurr))
                : (rentedBuffer = ArrayPool<byte>.Shared.Rent(totalLength));
```

At the end, you have to
```csharp
if (rentedBuffer is not null)
{
    // WriteDirectLarge(buffer.Slice(0, totalLength)); Check my next command
    while (!RespWriteUtils.WriteDirect(buffer.Slice(0, totalLength), ref dcurr, dend))
        SendAndReset();

    ArrayPool<byte>.Shared.Return(rentedBuffer);
}
else
{
    dcurr += totalLength;
}
```

Reply:
done